rosbag2/CMakeLists.txt

```diff
@@ -16,6 +16,11 @@ if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
 endif()
 
 option(DISABLE_SANITIZERS "disables the use of gcc saniztizers")
+# Only enable sanitizers on x64 architectures
+# https://github.com/google/sanitizers/issues/794
+if(NOT "${CMAKE_SIZEOF_VOID_P}" EQUAL "8")
```

Comment:
qq: Were you unable to get this running without the quotes? Ideally it would be int not str comaprison.
```
if(NOT CMAKE_SIZEOF_VOID_P EQUAL 8)
   set(DISABLE_SANITIZERS ON)
endif()
```

Reply:
absolutely right. https://github.com/ros2/rosbag2/pull/149/commits/3c1ab0dbbe7dd2bf2402688752a74e2c9ac061ee addresses this.